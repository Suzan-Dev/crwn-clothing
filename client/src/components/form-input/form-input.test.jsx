import React from 'react';
import { shallow } from '../../setupTests';
import FormInput from './form-input.component';

describe('Testing: FormInput component!', () => {
  const mockHandleChange = jest.fn();
  const mockLabel = 'Email';
  const mockValue = 'test@gmail.com';

  const mockProps = {
    handleChange: mockHandleChange,
    label: mockLabel,
    value: mockValue,
  };

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<FormInput {...mockProps} />);
  });

  test('should render FormInput component.', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('should call handleChange method when input changes.', () => {
    wrapper.find('.form-input').simulate('change');
    expect(mockHandleChange).toHaveBeenCalled();
  });

  test('should render label if there is a label.', () => {
    expect(wrapper.exists('.form-input-label')).toBeTruthy();
  });

  test('should not render label if there is no label.', () => {
    const mockProps2 = {
      ...mockProps,
      label: '',
    };
    const wrapper2 = shallow(<FormInput {...mockProps2} />);

    expect(wrapper2.exists('.form-input-label')).not.toBeTruthy();
  });
});
