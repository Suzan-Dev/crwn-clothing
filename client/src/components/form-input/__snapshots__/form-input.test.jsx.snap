// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing: FormInput component! should render FormInput component. 1`] = `
"<div className=\\"group\\">
  <input className=\\"form-input\\" onChange={[Function: mockConstructor]} value=\\"test@gmail.com\\" />
  <label className=\\"form-input-label shrink\\">
    Email
  </label>
</div>"
`;
